lib/src/parser/recursive.dart

```diff
@@ -531,6 +539,111 @@ class RecursiveAstParser {
+      NgToken innerValue) {
+    var text = innerValue.lexeme;
+    var absoluteTextOffset = innerValue.offset;
+    var tokens = new RegExp(r'({{)|(}})');
+    var mustaches = <InterpolationAst>[];
+    int seenOpenMustache;
+
+    var scanner = new StringScanner(text);
+    while (!scanner.isDone) {
+      var position = scanner.position;
+      var match = tokens.firstMatch(scanner.rest);
+      if (match != null) {
+        var matchPosition = position + match.start;
+
+        if (match.group(1) != null) {
+          // '{{' found.
+          if (seenOpenMustache == null) {
+            seenOpenMustache = matchPosition;
+          } else {
+            // Second '{{' found before '}}' closes it.
+            var firstMustacheBegin = absoluteTextOffset + seenOpenMustache;
+            exceptionHandler.handle(new AngularParserException(
+              NgParserWarningCode.UNTERMINATED_MUSTACHE,
+              firstMustacheBegin,
+              '{{'.length,
+            ));
+
+            mustaches.add(new InterpolationAst.parsed(
+              _source,
+              new NgToken.interpolationStart(firstMustacheBegin),
+              new NgToken.interpolationValue(
+                  firstMustacheBegin + '{{'.length,
+                  text.substring(
+                      seenOpenMustache + '{{'.length, matchPosition)),
+              new NgToken.generateErrorSynthetic(
+                  absoluteTextOffset + matchPosition,
+                  NgTokenType.interpolationEnd),
+            ));
+            seenOpenMustache = matchPosition;
+          }
+        }
+
+        if (match.group(2) != null) {
+          if (seenOpenMustache != null) {
+            var mustacheBegin = absoluteTextOffset + seenOpenMustache;
+            var mustacheEnd = absoluteTextOffset + matchPosition;
+            mustaches.add(new InterpolationAst.parsed(
+              _source,
+              new NgToken.interpolationStart(mustacheBegin),
+              new NgToken.interpolationValue(
+                  mustacheBegin + '{{'.length,
+                  text.substring(
+                      seenOpenMustache + '{{'.length, matchPosition)),
+              new NgToken.interpolationEnd(mustacheEnd),
+            ));
+            seenOpenMustache = null;
+          } else {
+            // Found '}}' before any '{{'
+            var mustacheBegin = absoluteTextOffset + position;
+            var mustacheEnd = absoluteTextOffset + matchPosition;
+            exceptionHandler.handle(new AngularParserException(
+              NgParserWarningCode.UNOPENED_MUSTACHE,
+              mustacheEnd,
+              '}}'.length,
+            ));
+            mustaches.add(new InterpolationAst.parsed(
+              _source,
+              new NgToken.generateErrorSynthetic(
+                  mustacheBegin, NgTokenType.interpolationStart),
+              new NgToken.interpolationValue(
+                  mustacheBegin, text.substring(position, matchPosition)),
```

Comment:
the issue is that parsing ` Bob` results in a dart Expression object that doesn't contain the space. So we can't suggest anything at that location. Only by parsing `( Bob)` could we suggest things before the space.

However, putting `(` around everything could lead to some very strange errors. For instance, `Bob) and stuff` would report an error for `and` instead of `)`.